lib/utils/getSelectorMeaningfulSource.mjs

```diff
@@ -0,0 +1,45 @@
+import selectorParser from 'postcss-selector-parser';
+
+/**
+ * Get the source of a selector, excluding surrounding whitespace and comments.
+ *
+ * @param {import('postcss-selector-parser').Selector} node
+ *
+ * @returns {{selectorStr: string, index: number, endIndex: number}}
+ */
+export default function getSelectorMeaningfulSource(node) {
+	const firstIndex = node.nodes.findIndex((x) => !selectorParser.isComment(x));
+	const lastIndex = node.nodes.findLastIndex((x) => !selectorParser.isComment(x));
```

Comment:
[suggest] Super minor. This may be a bit more readable, but you can keep it as-is. 👍🏼 

```js
import selectorParser from 'postcss-selector-parser';

const { isComment } = selectorParser;
```

```suggestion
	const firstIndex = node.nodes.findIndex((x) => !isComment(x));
	const lastIndex = node.nodes.findLastIndex((x) => !isComment(x));
```